tests/templates/kuttl/oidc/01_keycloak.yaml.j2

```diff
@@ -0,0 +1,215 @@
+            - name: KEYCLOAK_ADMIN_PASSWORD
+              value: admin
+          ports:
+{% if test_scenario['values']['oidc-use-tls'] == 'true' %}
+            - name: https
+              containerPort: 8443
+{% else %}
+            - name: http
+              containerPort: 8080
+{% endif %}
+          volumeMounts:
+            - name: realms
+              mountPath: /opt/keycloak/data/import
+            - name: tls
+              mountPath: /tls
+          readinessProbe:
+            httpGet:
+              path: /realms/$REALM
+{% if test_scenario['values']['oidc-use-tls'] == 'true' %}
+              port: 8443
+              scheme: HTTPS
+{% else %}
+              port: 8080
+              scheme: HTTP
+{% endif %}
+      volumes:
+        - name: realms
+          configMap:
+            name: $INSTANCE_NAME-realms
+        - ephemeral:
+            volumeClaimTemplate:
+              metadata:
+                annotations:
+                  secrets.stackable.tech/class: keycloak-tls
+                  secrets.stackable.tech/scope: service=$INSTANCE_NAME
+              spec:
+                accessModes:
+                - ReadWriteOnce
+                resources:
+                  requests:
+                    storage: "1"
+                storageClassName: secrets.stackable.tech
+                volumeMode: Filesystem
+          name: tls
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: $INSTANCE_NAME
+spec:
+  selector:
+    app: $INSTANCE_NAME
+  ports:
+    - protocol: TCP
+{% if test_scenario['values']['oidc-use-tls'] == 'true' %}
+      port: 8443
+{% else %}
+      port: 8080
+{% endif %}
+---
+apiVersion: authentication.stackable.tech/v1alpha1
+kind: AuthenticationClass
+metadata:
+  name: $INSTANCE_NAME-$NAMESPACE
+spec:
+  provider:
+    oidc:
+      hostname: $INSTANCE_NAME.$NAMESPACE.svc.cluster.local
+{% if test_scenario['values']['oidc-use-tls'] == 'true' %}
+      port: 8443
+{% else %}
+      port: 8080
+{% endif %}
+      rootPath: /realms/$REALM
+      scopes:
+        - email
+        - openid
+        - profile
+      principalClaim: preferred_username
+      providerHint: Keycloak
+      tls:
+        verification:
+          server:
+            caCert:
+              secretClass: keycloak-tls
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: keycloak
+---
+kind: Role
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: keycloak
+{% if test_scenario['values']['openshift'] == 'true' %}
+rules:
+- apiGroups: ["security.openshift.io"]
+  resources: ["securitycontextconstraints"]
+  resourceNames: ["privileged"]
+  verbs: ["use"]
+{% endif %}
+---
+kind: RoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: keycloak
+subjects:
+  - kind: ServiceAccount
+    name: keycloak
+roleRef:
+  kind: Role
+  name: keycloak
+  apiGroup: rbac.authorization.k8s.io
+---
+apiVersion: secrets.stackable.tech/v1alpha1
+kind: SecretClass
+metadata:
+  name: keycloak-tls
```

Comment:
Can we please suffix it with `$NAMESPACE`, so `name: keycloak-tls-$NAMESPACE`? Otherwise we have a race condition, as SecretClasses have cluster scope